Add Conditional Rendering to New Assets Page (#2720)
w3f/polkadot-wiki: docs/learn/learn-assets.md

---
id: learn-assets
title: Assets
sidebar_label: Assets
description: Learn about fungible and non-fungible assets in the Polkadot network.
slug: ../learn-assets
---

Assets in the {{ polkadot: Polkadot :polkadot }} {{ kusama: Kusama :kusama }} network can be 
represented on several chains. They can also take many forms, from a parachain's native token 
to on-chain representations of off-chain reserves. This page focuses on the latter, namely assets 
that would be issued by a creator (e.g. rights to audited, off-chain reserves held by the creator, 
or art issued as an NFT).

The {{ polkadot: [Statemint parachain](https://www.parity.io/blog/statemint-generic-assets-chain-proposing-a-common-good-parachain-to-polkadot-governance/) :polkadot }} 

> {{ kusama: Statemine :kusama }} hosts data structures and logic that

Reviewer: Blog post is still relevant.

```suggestion
{{ kusama: [Statemineparachain](https://www.parity.io/blog/statemint-generic-assets-chain-proposing-a-common-good-parachain-to-polkadot-governance/) :kusama }} hosts data structures and logic that
```

specialize in the creation, management, and use of assets in the {{ polkadot: Polkadot :polkadot }} 
{{ kusama: Kusama :kusama }} network. Although other
parachains can host applications dealing with assets on {{ polkadot: Statemint, Statemint :polkadot }} 
{{ kusama: Statemine, Statemine :kusama }} can be thought of as the "home base" of assets in the 
network.

{{ polkadot: Statemint uses DOT as its native token :polkadot }} 

> {{ kusama: Statemine uses KSM as its native token :kusama }}. The chain yields its governance to

Reviewer: ```suggestion
{{ polkadot: Statemint uses DOT :polkadot }} 
{{ kusama: Statemine uses KSM :kusama }} as its native token. The chain yields its governance to
```

> its respective Relay Chain, and has no inflation or era-based rewards for collators. Collators

Reviewer: ```suggestion
its parent Relay Chain, and has no inflation or era-based rewards for collators. Collators
```

receive a portion of transaction fees. As a
[common good parachain](https://polkadot.network/blog/common-good-parachains-an-introduction-to-governance-allocated-parachain-slots/), {{ polkadot: Statemint 
has a trusted relationship with the Relay Chain, 

> and as such, can teleport DOT between itself and its Relay Chains. That is, DOT on Statemint is 

Reviewer: ```suggestion
and as such, can teleport DOT between itself and the Relay Chain. That is, DOT on Statemint is 
```

just as good as DOT on the Relay Chain. :polkadot }}
{{ kusama: Statemine has a trusted relationship with the Relay Chain, 

> and as such, can teleport KSM between itself and its Relay Chains. That is, KSM on Statemine is 

Reviewer: ```suggestion
and as such, can teleport KSM between itself and the Relay Chain. That is, KSM on Statemine is 
```

just as good as KSM on the Relay Chain. :kusama }}

{{ polkadot: Statemint :polkadot }} {{ kusama: Statemine :kusama }} does not support smart contracts. 
See the [Advanced](#advanced-techniques) section at the bottom for discussion on using proxy and multisig 
accounts to replicate oft used contract logic.

## Fungible Assets

Fungible assets are those that are interchangeable, i.e. one unit is equivalent to any other unit
for the purposes of claiming the underlying item. {{ polkadot: Statemint :polkadot }} 
{{ kusama: Statemine :kusama }} represents fungible assets in the 
Assets pallet. For those familiar with the ERC20 standard, this pallet presents a similar interface.
However, the logic is encoded directly in the chain's runtime. As such, operations are not gas
metered and instead are benchmarked upon every release, leading to efficient execution and stable
transaction fees.

### Creation and Management

Anyone on the network can create assets on {{ polkadot: Statemint as long as they can reserve the 
required deposit of 100 DOT on Statemint. :polkadot }} {{ kusama: Statemine as long as they can reserve 

> the required deposit of 1 KSM on Statemine. :kusama }}. The network reserves the deposit on creation.

Reviewer: ```suggestion
Anyone on the network can create assets on {{ polkadot: Statemint, as long as they can reserve the 
required deposit of 100 DOT. :polkadot }} {{ kusama: Statemine, as long as they can reserve 
the required deposit of 1 KSM. :kusama }}. The network reserves the deposit on creation.
```

---
id: learn-assets
title: Assets
sidebar_label: Assets

> description: Learn about fungible and non-fungible assets in the Polkadot ecosystem.

Reviewer: This should definitely stay "network"

Author: Updated

- Admin
- Freezer

The owner has the ability to set the accounts responsible for the other three roles, as well as set 
asset metadata (e.g. name, symbol, decimals). The issuer can mint and burn tokens to/from addresses 
of their choosing. The freezer can freeze assets on target addresses or the entire asset class. The 
admin can make force transfers as well as unfreeze accounts of the asset class. **Always refer to the
[reference documentation](https://crates.parity.io/pallet_assets/index.html) for certainty on
privileged roles.**

An asset's details contain one field not accessible to its owner or admin team, that of asset
sufficiency. Only the network's governance mechanism can deem an asset as *sufficient.* A balance of
a non-sufficient asset (the default) can only exist on already-existing accounts. That is, a user
could not create a new account on-chain by transferring an insufficient asset to it; the account
must already exist by having more than the existential deposit {{ polkadot: in DOT :polkadot }} 

> {{ kusama: in KSM :kusama }}. However, assets deemed *sufficient* can instantiate accounts. In the 

Reviewer: ```suggestion
must already exist by having more than the existential deposit {{ polkadot: in DOT :polkadot }} 
{{ kusama: in KSM :kusama }} (or a sufficient asset). However, assets deemed *sufficient* can instantiate accounts. In the 
```

represents NFTs in the [Uniques pallet](https://crates.parity.io/pallet_uniques/index.html).

Similar to the Assets pallet, this functionality is encoded into the chain. Operations are
benchmarked prior to each release in lieu of any runtime metering, ensuring efficient execution and
stable transaction fees.

### Creation and Management

{{ polkadot: Anyone on the network can create an asset class, as long as they reserve the required 
deposit of 100 DOT on Statemint :polkadot }} {{ kusama: Anyone on the network can create an asset class, 
as long as they reserve the required deposit of 1 KSM on Statemine :kusama }}. Creating instances of a 
class also requires a per-instance deposit, unless the chain's governance designates the class as 
"free holding", allowing the class to mint more instances without deposit. The creator must specify a 
`ClassId`, which, like its cousin `AssetId`, should be the canonical identifier for the class.

The creator can also specify the same privileged roles of Owner, Admin, Issuer, and Freezer.

Asset classes and instances can have associated metadata. The metadata is an array of data that the

> class Owner can add on-chain, for example, a link to an IPFS hash or other off-chain hosting services.

Reviewer: The "other" implies that it's one of many.
```suggestion
class Owner can add on-chain, for example, a link to an IPFS hash or other off-chain hosting service.
```
